app/services/cache_service.rb

```diff
@@ -43,7 +50,21 @@ def listing_unchanged?(prev_cached_listing, fresh_listing)
       (prev_cached_listing['LastModifiedDate'] == fresh_listing['LastModifiedDate'])
   end
 
+  def listing_images_unchanged?(prev_cached_listing, fresh_listing)
+    prev_cached_listing_images = prev_cached_listing&.dig('Listing_Images')
+    fresh_listing_images = fresh_listing_images&.dig('Listing_Images')
+
+    return true if fresh_listing_images.blank?
+
+    fresh_li_slice = fresh_listing_images&.map { |li| li.slice('Id', 'Image_URL') }
+    prev_li_slice = prev_cached_listing_images&.map { |li| li.slice('Id', 'Image_URL') }
+    Rails.logger.info("fresh #{fresh_li_slice}, prev #{prev_li_slice}")
```

Comment:
This was copy-pasted from my sample code, I would adjust it to be more descriptive, and/or remove it if you see no benefit.

Also, for some reason I never saw this log when I was testing in the PR environment, not sure why.